### ISSUE TITLE
Isofox 1.5 crash

### ISSUE THREAD

**Issue by toddajohnson** [hartwigmedical/hmftools#292]
Downloaded isofox_v1.5.jar, and with java -jar isofox_v1.5.jar, it returns "15:24:05 - [INFO ] - Isofox version: 1.4".

Also, while Isofox 1.4 runs to completion, but with the same commands:
`	module use /usr/local/package/modulefiles/
	module load java/11
	functions_to_run="TRANSCRIPT_COUNTS;ALT_SPLICE_JUNCTIONS;RETAINED_INTRONS;FUSIONS"
	
	export _JAVA_OPTIONS="-Xmx${jvm_mem}"

	java -jar ${ISOFOX_JAR} \
		-sample ${WGS_sample_id} \
		-functions  ${functions_to_run}\
		-bam_file ${bam_path} \
		-ref_genome ${REFERENCE_GENOME_FA} \
		-ref_genome_version ${REFERENCE_GENOME} \
		-ensembl_data_dir ${ensembl_cache_dir} \
		-output_dir ${output_path} \
		-read_length ${read_length} \
		-exp_counts_file "${ref_dir}/dbs/isofox/read_${read_length}_exp_counts.csv" \
		-exp_gc_ratios_file "${ref_dir}/dbs/isofox/read_${read_length}_exp_gc_ratios.csv" \
		-long_frag_limit 550 \
		-known_fusion_file "${ref_dir}/dbs/knowledgebases/known_fusion_data.38_v3.csv" \
		-threads ${threads} `

Isofox 1.5 crashed with:

14:49:49 - [INFO ] - Isofox version: 1.4
14:49:49 - [INFO ] - running function(s): TRANSCRIPT_COUNTS
14:49:49 - [INFO ] - running function(s): ALT_SPLICE_JUNCTIONS
14:49:49 - [INFO ] - running function(s): RETAINED_INTRONS
14:49:49 - [INFO ] - running function(s): FUSIONS
14:49:49 - [INFO ] - loaded known fusion data: PROMISCUOUS_5(31), IG_PROMISCUOUS(1), PROMISCUOUS_3(32), KNOWN_PAIR(402), EXON_DEL_DUP(12), IG_KNOWN_PAIR(18)
14:49:54 - [INFO ] - fragment lengths(75, 100, 125, 150, 200, 250, 300, 400, 550) set from expected transcript counts file
14:50:00 - [INFO ] - loaded 26592 gene expected counts from file(/home/tjohnson/reference/HMF/38/dbs/isofox/read_101_exp_counts.csv)
14:50:04 - [INFO ] - loaded 244496 transcript expected GC ratios from file(/home/tjohnson/reference/HMF/38/dbs/isofox/read_101_exp_gc_ratios.csv)
14:50:04 - [INFO ] - sample(RK001_R) running RNA analysis
14:50:31 - [INFO ] - max observed read length(101) set
14:50:31 - [INFO ] - fragmentLength(75) frequency(6182)
14:50:31 - [INFO ] - fragmentLength(100) frequency(15080)
14:50:31 - [INFO ] - fragmentLength(125) frequency(84764)
14:50:31 - [INFO ] - fragmentLength(150) frequency(317100)
14:50:31 - [INFO ] - fragmentLength(200) frequency(286806)
14:50:31 - [INFO ] - fragmentLength(250) frequency(107705)
14:50:31 - [INFO ] - fragmentLength(300) frequency(47250)
14:50:31 - [INFO ] - fragmentLength(400) frequency(12599)
14:50:31 - [INFO ] - fragmentLength(550) frequency(875)
14:50:31 - [INFO ] - PerfStats: name(FragLengthDist) intervals(13305) total(148.045 avg=0.011 max=10.503)
Exception in thread "main" java.lang.NullPointerException
	at com.hartwig.hmftools.isofox.fusion.ChimericReadTracker.registerKnownFusionPairs(ChimericReadTracker.java:140)
	at com.hartwig.hmftools.isofox.BamFragmentAllocator.registerKnownFusionPairs(BamFragmentAllocator.java:880)
	at com.hartwig.hmftools.isofox.ChromosomeTaskExecutor.<init>(ChromosomeTaskExecutor.java:112)
	at com.hartwig.hmftools.isofox.Isofox.allocateBamFragments(Isofox.java:195)
	at com.hartwig.hmftools.isofox.Isofox.runAnalysis(Isofox.java:147)
	at com.hartwig.hmftools.isofox.Isofox.main(Isofox.java:491)

**Comment by charlesshale** [comment]
That NPE implies that one of the genes in the known fusion file is not in the Ensembl data cache. I've just added protection against that but would be good to check both files are the latest. I've been running with V38 files too and not encountered that.

Updated JAR (rc1), also with the version now logging as 1.5:

https://github.com/hartwigmedical/hmftools/releases/tag/isofox-v1.5

**Comment by toddajohnson** [comment]
Thanks, Charles!
I updated known_fusion_data and the Ensembl data cache and it ran fine.
I had been using an Ensembl 104 cache that I created earlier last year based on your gene-utils code, but I have not updated since then (thinking that it was somewhat static), and also I see now in gene-utils that you update the cache based on current HGNC names.  For several hundred genes the names have changed since 104 was released, which created the discrepancy.  Probably, the NPE was a good wakeup call!
